### ISSUE TITLE
A minor issue in CSVDiffer

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#12431]
## Rationale

I changed recently in my code, added few more postprocessors and then got a KeyError in few CSV diff tests at line https://github.com/idaholab/moose/blob/devel/python/TestHarness/CSVDiffer.py#L112. Looks to me the differ allows the numbers of columns are different from the gold file and the current file. Thus the fix is just replace `keys1` to `keys2`

## Description

As Rationale. If this is confirmed, I can send in a small PR to close this.

## Impact
Minor because typically users want to regold the file. Just to make things logically right and avoid the key error.

**Comment by permcody** [comment]
This PR is fine, but we are going to make the CSVDiffer behave more like the Exodiff utility for consistency. I'm getting a little tired of piling on small hacks to make it do what we all need it to do. There's just not enough flexibility with the options we currently have. I suspect that this use case will likely need a "custom compare" file just like we require for Exodiff now when the new version goes in. 

In other words, when you ask for a diff, you'll get a diff of all columns and rows so having too few or too many would be a detected as a diff. If you want only a subset of columns (or rows) compared, you'd have to create a separate file to tell the utility how to behave. 

I just want to keep you informed.
